algolia/algoliasearch-sfcc-b2c
# fix: handle emojis errors in delta jobs

When some product attribute contain an emoji (e.g. 🐣), SFCC Delta export fails to generate a valid XML: the XML contains those characters: `&#55357;&#56355;`

To solve this problem, the **"Filter Invalid XML Characters during Export"** feature must be enabled in _Administration > Global Preferences > Import & Export_. When enabled, the emojis are correctly exported. (I found that thanks to [this Medium post](https://3131ritikyadav.medium.com/character-reference-55357-is-an-invalid-xml-character-eis-e2c3becec974))

This PR makes to following improvements to the Delta export job:
- Report the actual error in the Business Manager. Previously we were only reporting a generic _"Error reading from file..."_ message
  - The message also let users know about the "Filter Invalid XML Characters during Export" feature
- Move the problematic delta export files to a `_failed` directory. This permit to not block subsequent jobs. Currently, the file needs to be deleted manually

## How to test

- Insert an emoji in a product description
- Run the delta export job
- The error should be properly reported in the logs and in the Business Manager
- The delta export files should be moved to `/IMPEX/src/platform/outbox/algolia/productDeltaExport/_failed/`

![image](https://github.com/user-attachments/assets/557cf7ad-7801-42fd-8724-df7b43a46173)

- Then enable the "Filter Invalid XML Characters during Export" feature
- Insert another emoji in a product description
- Run the delta export job, which should complete successfully
